### ISSUE TITLE
A _repr_html_ for a matchpy expression.

### ISSUE THREAD

**Issue by tonyfast** [Quansight-Labs/uarray#20]
In the web view, a pure html representation might be better for the `matchpy...Expressions`

```python
 import pprint, black, vdom, matchpy, IPython; IPython.get_ipython().display_formatter.formatters['text/html'].for_type(
        matchpy.expressions.expressions.Expression, lambda str: vdom.pre(vdom.code((black).format_str(pprint.pformat(str), 100)))._repr_html_()
    )
```

This snippet is meant to be registered in a notebook to demonstrate the idea.

**Comment by tonyfast** [comment]
Here is a sample view:

![image](https://user-images.githubusercontent.com/4236275/47514945-8fb7bf00-d84f-11e8-848b-f8190e57ad6b.png)

**Comment by saulshanabrook** [comment]
@pearu Yes that is a good idea. The `__str__` version does this partially, but it doesn't do any indenting. It might be nice to have a `__str__` pprint version as well as a `__repr__` pprint version

**Comment by saulshanabrook** [comment]
I have this kind of pretty printing now working in jupyter notebooks: https://github.com/Quansight-Labs/uarray/blob/master/uarray/uarray/printing.py

**Comment by saulshanabrook** [comment]
Ah nice thanks for that code snippet. Reopening this to adopt that
